WIP - tracker-web refactor

danieljoonlee opened the pull request:
@dhoward @alexpeachey @ssebelius 

Based off of our last conversation I whipped up some of the suggestions.  Rather than creating a lot of the structure for the interviewees, now it's all stuffed into one file, so you can view their approaches.  We don't really use redux, so I removed that for now.  Seemed overkill for a simple to do app.  If you guys think this is going towards the right direction, I can fix the cucumber, lint, add styling, make the mock up, etc.  Then we can discuss what a realistic amount of work might look like in 4 hours.

dhoward commented:
Nice! I like the new direction. Just have a few comments.

1. Now that redux is gone, it doesn't look like `initialState.js` is being used anywhere, can that go too?

2. I don't think we ask them to do anything environment-specific, should we get rid the various webpack config files and condense them into one?

3. I know this is WIP but it looks like the feature tests were using some of the Redux stuff so they may need to be reworked a little to make sure they run.

4. Should we put a couple sample unit tests to get people on the right track there?

danieljoonlee commented:
@dhoward yeah I can get rid of the `initialState.js` along with fixing up the feature tests.  I agree with the environment-specific note, we can probably clean that up as well.  

For the sample unit tests that sounds good to me.  What do you guys think? @ssebelius and @alexpeachey

danieljoonlee commented:
@alexpeachey @ssebelius 

Currently `cucumber` is using `enzyme`, which was one of the issues we had before.  Should we remove the `cucumber` portion, or I can implement the current stuff we're using with `nightwatch + cucumber`

alexpeachey commented:
Due to the simple interactions enzyme works ok on this app, but yeah, if we want to add more or make it more "real" might need nightwatch

dhoward commented:
I would be inclined to say that figuring out cucumber + selenium + nightwatch might fall outside the scope of a 4-hour-ish coding challenge, but ya I would probably favor that over the enzyme solution since at least it mimics our actual test setup

danieljoonlee commented:
@dhoward @alexpeachey @ssebelius 

Made a lot of the PR suggestions along with in-person meetings.  Removed cucumber, condensed webpack configs, added lint and test watchers, added a couple of unit tests, and replaced the design with the balsamic? mock up, along with the readme updates.

dhoward commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
 .DS_Store
 npm-debug.log
 node_modules/
+.idea/
```
How about you just use Sublime like a real front end developer?

dhoward commented on README.md:
```diff
@@ -12,25 +12,14 @@ This application is very incomplete and as such has plenty of room for you to ad
-* Green: #3EC556
-* Blue: #3E74C5
-
+The project has been setup with React and Material-UI and we would like you to make the dashboard similar to the picture.
```
Maybe point them to it? "similar to the picture in the /design folder"
